src/Mod/Part/App/TopoShapeEdgePyImp.cpp

```diff
@@ -1045,6 +1049,42 @@ Py::Boolean TopoShapeEdgePy::getDegenerated(void) const
     return Py::Boolean(ok ? true : false);
 }
 
+PyObject* TopoShapeEdgePy::curveOnSurface(PyObject *args)
+{
+    int idx;
+    if (!PyArg_ParseTuple(args, "i", &idx))
+        return 0;
+
+    try {
+        TopoDS_Edge edge = TopoDS::Edge(getTopoShapePtr()->getShape());
+        Handle(Geom2d_Curve) curve;
+        Handle(Geom_Surface) surf;
+        TopLoc_Location loc;

```

Comment:
I don't know what to do with this Location ?
Should I convert it to Placement, or Matrix, and return this also ?